### PR TITLE
Document adoption of Conventional Commits.

### PR DESCRIPTION
**Pull request by apetro**
Document in contributing guidance to use Conventional Commits, while emphasizing that contributing guidance is not intended to be an insurmountable barrier to contribution.

----

Contributor License Agreement adherence:

- [x] This Contribution is under the terms of Individual [Contributor License Agreements][] (and also Corporate Contributor License Agreements to the extent applicable) appearing in the [Apereo CLA roster][].

[Apereo CLA roster]: http://licensing.apereo.org/completed-clas
[Contributor License Agreements]: https://www.apereo.org/licensing/agreements

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -43,15 +46,49 @@ npm run lint-md
...
+We have not yet formalized scopes. In practice scopes used so far include:
+
++ `changelog`
++ `ci`
++ `dependency`
++ `dependency upgrade`
++ `github`
++ `md`
++ `npm`
++ `package`
```
Comment (ChristianMurphy) [inline .github/CONTRIBUTING.md]: :question: what is the difference between `dependency`, `dependency upgradle`, and `package`?

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -43,15 +46,49 @@ npm run lint-md
...
+
+We use [the change types Angular uses][].
+
+
+### Scope
+
+We have not yet formalized scopes. In practice scopes used so far include:
+
++ `changelog`
++ `ci`
```
Comment (ChristianMurphy) [inline .github/CONTRIBUTING.md]: :question: ci is already a type, do we also want it to be a scope?

Comment (apetro) [reply .github/CONTRIBUTING.md]: Nope. :)

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (apetro) [reply .github/CONTRIBUTING.md]: Dunno. Colorable argument for collapsing these. I figured the first step towards that was writing down what we've got so far.

**Pull request merged**